CMakeLists.txt

```diff
@@ -705,7 +708,13 @@ if(HERMES_IS_ANDROID)
   find_package(fbjni REQUIRED CONFIG)
 endif()
 
-set(CMAKE_CXX_STANDARD 17)
+if("${CMAKE_CXX_COMPILER_ID}" MATCHES "MSVC")
+  # MSCV requires c++20
```

Comment:
Exactly like you said. MSVC requires C++20 because of the use of designated initialisers in the codebase. Here are some sample of logs from CI:
```
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(161,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(163,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(165,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(167,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(169,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(171,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(173,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(175,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(177,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(179,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(181,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(158,5): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(259,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(261,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(263,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(265,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(267,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(269,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(271,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(273,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(275,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(277,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(279,14): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
D:\tmp\hermes\hermes\lib\AST\NativeContext.cpp(256,5): error C7555: use of designated initializers requires at least '/std:c++20' [D:\tmp\hermes\build_release\lib\AST\hermesAST_obj.vcxproj]
```